internal/services/tenant_isolation_policy/resource_tenant_isolation_policy_test.go

```diff
@@ -0,0 +1,605 @@
+					resource.TestCheckResourceAttr("powerplatform_tenant_isolation_policy.test", "allowed_tenants.#", "1"),
+				),
+			},
+			{
+				Config: `
+				resource "powerplatform_tenant_isolation_policy" "test" {
+					is_disabled = true
+					allowed_tenants = toset([
+						{
+							tenant_id = "11111111-1111-1111-1111-111111111111"
+							inbound = true
+							outbound = true
+						},
+						{
+							tenant_id = "22222222-2222-2222-2222-222222222222"
+							inbound = true
+							outbound = false
+						}
+					])
+				}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("powerplatform_tenant_isolation_policy.test", "is_disabled", "true"),
+					resource.TestCheckResourceAttr("powerplatform_tenant_isolation_policy.test", "allowed_tenants.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+// TestAccTenantIsolationPolicy_Validate_Delete tests the removal of all allowed tenants from a policy.
+// It verifies:
+// 1. Initial creation of a policy with one allowed tenant
+// 2. Updating the policy to remove all allowed tenants
+// 3. The policy exists but has an empty allowed_tenants list.
+func TestAccTenantIsolationPolicy_Validate_Delete(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: mocks.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: `
+				resource "powerplatform_tenant_isolation_policy" "test" {
+					is_disabled = false
+					allowed_tenants = toset([
+						{
+							tenant_id = "11111111-1111-1111-1111-111111111111"
+							inbound = true
+							outbound = true
+						}
+					])
+				}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("powerplatform_tenant_isolation_policy.test", "allowed_tenants.#", "1"),
+				),
+			},
+			{
+				Config: `
+				resource "powerplatform_tenant_isolation_policy" "test" {
+					is_disabled = false
+					allowed_tenants = toset([])
+				}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("powerplatform_tenant_isolation_policy.test", "allowed_tenants.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+// TestUnitTenantIsolationPolicyResource_Validate_Create tests the creation of a tenant isolation policy
+// using mocked API responses. It verifies:
+// 1. The API calls are made with correct parameters
+// 2. The resource correctly processes the API response
+// 3. The resource state matches the expected configuration with one allowed tenant.
+func TestUnitTenantIsolationPolicyResource_Validate_Create(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	setupTenantHttpMocks()
+
+	// Setup PUT responder for creating the policy.
+	httpmock.RegisterResponder("PUT", fmt.Sprintf("https://api.bap.microsoft.com/providers/PowerPlatform.Governance/v1/tenants/%s/tenantIsolationPolicy", testTenantID),
+		func(req *http.Request) (*http.Response, error) {
+			return httpmock.NewStringResponse(http.StatusOK, `{
+                "properties": {
```

Comment:
THis json response should be in a separate file in a test/Validate_Create/ folder